### DISCUSSION TITLE
Issues with loading mason-registry behind proxy

### DISCUSSION THREAD

**Discussion by 1g0r** [mason-org/mason.nvim#1735] [Q&A]
Hello!

I'm using mason on windows box. and have encountered a problem with downloading mason-registry using :Mason command.
I have no issues with using git and downloading various plugins using :Lazy. But mason can't download lsp servers

I've investigated the problem and I've changed mason's  file https://github.com/williamboman/mason.nvim/blob/main/lua/mason-core/fetch.lua  locally for mason to use proper proxy settings like so
```lua
    if platform.is.win then
        local header_entries = _.join(
            "; ",
            _.map(function(pair)
                return ("%q=%q"):format(pair[1], pair[2])
            end, _.to_pairs(opts.headers))
        )
        local headers = ("-Headers @{%s}"):format(header_entries)
        if opts.out_file then
            platform_specific = function()
                return powershell.command(
HERE:              ([[iwr %s -TimeoutSec %s -UseBasicParsing -Method %q -Uri %q %s -OutFile %q -Proxy %q -ProxyUseDefaultCredentials;]]):format(
                        headers,
                        TIMEOUT_SECONDS,
                        opts.method,
                        url,
                        opts.data and ("-Body %s"):format(opts.data) or "",
                        opts.out_file,
HERE:               'http://corporate-proxy:xxx' -- hardcoded proxy
                    )
                )
            end
        else
            platform_specific = function()
                return powershell.command(
HERE:               ([[Write-Output (iwr %s -TimeoutSec %s -Method %q -UseBasicParsing %s -Uri %q -Proxy %q -ProxyUseDefaultCredentials).Content;]]):format(
                        headers,
                        TIMEOUT_SECONDS,
                        opts.method,
                        opts.data and ("-Body %s"):format(opts.data) or "",
                        url,
HERE:               'http://corporate-proxy:xxx' -- hardcoded proxy
                    )
                )
            end
        end
    end
```

See the words `ProxyUseDefaultCredentials` and `Proxy`.  I have to hardcode the proxy due to the lack of understanding of how to get the value from opts. My solution works just fine till update mason using :Lazy

Is there a batter way to configure proxy usage with default credentials using lua configuration? I haven't found any in documentation.

Thanks a lot!

**Comment by zephod77** [comment]
Any progress on this? 
I have the same error. See screenshot. 
The mason logs seem to indicate it has to do with something called omnisharp. See second screenshot. I guess this is some C# module but I don't have it on my system. I am also behind a proxy which changed recently so please let us know if there are some configuration options needed to get past the proxy. The exit code 5 (SSL verification error) from wget is the clue.

![Mason_error](https://github.com/user-attachments/assets/1d1194c1-c4c1-4b50-b210-381cf03fc0a3)
![Mason_log](https://github.com/user-attachments/assets/0003b11c-720a-4159-9280-6b3a83fe9f4b)

**Comment by 1g0r** [comment]
Hello!
Is there any news on this issue? I have to change the script manually every time I update Mason
